pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -271,22 +267,29 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 		}
 	}
 
-	lcSpec := applicationLayer.Spec.LogCollection
+	// prior client.Patch may nil out missing fields (in cases of outdated crd spec).
+	// We need to re-fill out defaults again. We don't need to do this if CRD is updated to match.
+	updateApplicationLayerWithDefaults(alp)
+
+	lcSpec := alp.Spec.LogCollection
 	config := &applicationlayer.Config{
 		PullSecrets:            pullSecrets,
 		Installation:           installation,
 		OsType:                 rmeta.OSTypeLinux,
-		WAFEnabled:             r.isWAFEnabled(&applicationLayer.Spec),
-		LogsEnabled:            r.isLogsCollectionEnabled(&applicationLayer.Spec),
-		ALPEnabled:             r.isALPEnabled(&applicationLayer.Spec),
+		WAFEnabled:             r.isWAFEnabled(&alp.Spec),
+		LogsEnabled:            r.isLogsCollectionEnabled(&alp.Spec),
+		ALPEnabled:             r.isALPEnabled(&alp.Spec),
 		LogRequestsPerInterval: lcSpec.LogRequestsPerInterval,
 		LogIntervalSeconds:     lcSpec.LogIntervalSeconds,
 		ModSecurityConfigMap:   modSecurityRuleSet,
+		UseRemoteAddressXFF:    alp.Spec.EnvoySettings.UseRemoteAddress,
+		NumTrustedHopsXFF:      alp.Spec.EnvoySettings.XFFNumTrustedHops,
+		SkipAppendXFF:          alp.Spec.EnvoySettings.SkipXFFAppend,
```

Comment:
This should be removed I think, since we were cutting down from the 3 new fields to only 2.